test/system/libpod_suite_test.go

```diff
@@ -0,0 +1,696 @@
+			matches = true
+			greps = append(greps, line)
+		}
+	}
+	return matches, greps
+}
+
+// ErrorToString formats session stderr to string
+func (s *PodmanSession) ErrorToString() string {
+	fields := strings.Fields(fmt.Sprintf("%s", s.Err.Contents()))
+	return strings.Join(fields, " ")
+}
+
+// ErrorToStringArray returns the stderr output as a []string
+// where each array item is a line split by newline
+func (s *PodmanSession) ErrorToStringArray() []string {
+	output := fmt.Sprintf("%s", s.Err.Contents())
+	return strings.Split(output, "\n")
+}
+
+// IsJSONOutputValid attempts to unmarshal the session buffer
+// and if successful, returns true, else false
+func (s *PodmanSession) IsJSONOutputValid() bool {
+	var i interface{}
+	if err := json.Unmarshal(s.Out.Contents(), &i); err != nil {
+		fmt.Println(err)
+		return false
+	}
+	return true
+}
+
+func (s *PodmanSession) WaitWithDefaultTimeout() {
+	s.Wait(defaultWaitTimeout)
+}
+
+// SystemExec is used to exec a system command to check its exit code or output
+func SystemExec(command string, args []string) *PodmanSession {
+	c := exec.Command(command, args...)
+	session, err := gexec.Start(c, GinkgoWriter, GinkgoWriter)
+	if err != nil {
+		Fail(fmt.Sprintf("unable to run command: %s %s", command, strings.Join(args, " ")))
+	}
+	return &PodmanSession{session}
+}
+
+// CreatePod creates a pod with no infra container
+// it optionally takes a pod name
+func (p *PodmanTest) CreatePod(name string) (*PodmanSession, int, string) {
+	var podmanArgs = []string{"pod", "create", "--infra=false", "--share", ""}
+	if name != "" {
+		podmanArgs = append(podmanArgs, "--name", name)
+	}
+	session := p.Podman(podmanArgs)
+	session.WaitWithDefaultTimeout()
+	return session, session.ExitCode(), session.OutputToString()
+}
+
+//RunTopContainer runs a simple container in the background that
+// runs top.  If the name passed != "", it will have a name
+func (p *PodmanTest) RunTopContainer(name string) *PodmanSession {
+	var podmanArgs = []string{"run"}
+	if name != "" {
+		podmanArgs = append(podmanArgs, "--name", name)
+	}
+	podmanArgs = append(podmanArgs, "-d", ALPINE, "top")
+	return p.Podman(podmanArgs)
+}
+
+func (p *PodmanTest) RunTopContainerInPod(name, pod string) *PodmanSession {
+	var podmanArgs = []string{"run", "--pod", pod}
+	if name != "" {
+		podmanArgs = append(podmanArgs, "--name", name)
+	}
+	podmanArgs = append(podmanArgs, "-d", ALPINE, "top")
+	return p.Podman(podmanArgs)
+}
+
+//RunLsContainer runs a simple container in the background that
+// simply runs ls. If the name passed != "", it will have a name
+func (p *PodmanTest) RunLsContainer(name string) (*PodmanSession, int, string) {
+	var podmanArgs = []string{"run"}
+	if name != "" {
+		podmanArgs = append(podmanArgs, "--name", name)
+	}
+	podmanArgs = append(podmanArgs, "-d", ALPINE, "ls")
+	session := p.Podman(podmanArgs)
+	session.WaitWithDefaultTimeout()
+	return session, session.ExitCode(), session.OutputToString()
+}
```

Comment:
This is very e2e test specific. They add functions for highly used containers in the test. Will left this to e2e only. I don't think system test need this.

Reply:
I disagree, I think have generic function will be very useful in system-test because this is most common pattern:

1. Set `subCommand`, ``podmanArgs`` tuple array
2. for each ``cmd``, ``args``
    a.  execute ``podman <cmd> <args>`` 
    b. check result is good

For example:
1. podman pull old_image
2. podman build Dockerfile (FROM: new_image)
3. podman run built_image
4. verify new_image was pulled
5. verify container use new_image and not old_image

Have separate 'do something' and 'check result' for every item will be much duplicate test-code.